### PR TITLE
Introduced ClickObserver.

### LINKED ISSUE #577: Introduce click observer
Simple click observer is needed by link feature.

### PR DESCRIPTION
**Pull request by oskarwrobel**
Fixes #577.

### CODE REVIEW

File: src/view/observer/clickobserver.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import DomEventObserver from './domeventobserver.js';
+
+/**
+ * {@link engine.view.Document#click Click} event observer.
+ *
+ * Note that this observer is not attached by the {@link engine.view.Document} and is not available by default.
```
Comment (Reinmar) [inline src/view/observer/clickobserver.js]: If we're saying this it would be much more useful if we said how then to enable this observer.

Comment (oskarwrobel) [inline src/view/observer/clickobserver.js]: Right, done.

Comment (oleq) [inline src/view/observer/clickobserver.js]: The only thing missing here is rich documentation, similar to [which comes with the `KeyObserver`](https://github.com/ckeditor/ckeditor5-engine/blob/t/577/src/view/observer/keyobserver.js#L39-L65).

**Comment by oleq** [comment]
Docs needed. Besides that, it looks yummy.

**Pull request merged**